### PR TITLE
Conditionally add ROCPrim identifier for ROCm

### PR DESCRIPTION
**Pull request by hubertlu-tw**
The logs of CuPy builds with this fix on ROCm5.2 (`rocm5.2_ubuntu20.04_py3.7_pytorch_staging`) and ROCm5.4 (c`ompute-artifactory.amd.com:5000/rocm-plus-docker/framework/compute-rocm-dkms-no-npi-hipclang:10847_ubuntu20.04_py3.7_pytorch_rocm5.4_internal_testing_2b17d9a`) can be found in the following attachments:
- [cupy_fix_rocm5.2.txt](https://github.com/ROCmSoftwarePlatform/cupy/files/9701389/cupy_fix_rocm5.2.txt)
- [cupy_fix_rocm5.4.txt](https://github.com/ROCmSoftwarePlatform/cupy/files/9701390/cupy_fix_rocm5.4.txt)

### CODE REVIEW

**Review by AdrianAbeyta — APPROVED** [review]
LGTM!

File: cupy/cuda/cupy_thrust.cu [hunk 1]
```diff
@@ -349,7 +349,17 @@ struct _sort {
             transform(cuda::par(alloc).on(stream_),
                       make_counting_iterator<size_t>(0),
                       make_counting_iterator<size_t>(size),
+                      #ifdef __HIP_PLATFORM_HCC__
```
Comment (amathews-amd) [inline cupy/cuda/cupy_thrust.cu:350]: @AdrianAbeyta , once upstream PR is done, make sure to check this in the next merge

**Review by amathews-amd — APPROVED** [review]
LGTM

**Comment by hubertlu-tw** [comment]
Please find the log of ROCm5.2 with the new patch which removed the if-statement for different ROCm releases here:
[cupy_new_fix_rocm5.2.txt](https://github.com/ROCmSoftwarePlatform/cupy/files/9701730/cupy_new_fix_rocm5.2.txt)

**Pull request closed without merging**